torch/distributed/fsdp/_common_utils.py

```diff
@@ -272,13 +274,14 @@ def _apply_to_modules(
     to remove the prefix.
     """
 
-    def f(module: torch.nn.Module, prefix: str, *args, **kwargs):
+    def f(module: torch.nn.Module, prefix: str, tree_level: int, *args, **kwargs):
         # Call the module function before recursing over children (pre-order)
-        module_fn(module, prefix, *args, **kwargs)
+        module_fn(module, prefix, tree_level, *args, **kwargs)
         for submodule_name, submodule in module.named_children():
             if submodule is None:
                 continue
             new_prefix = prefix + submodule_name + "."
+            new_tree_level = tree_level + 1
```

Comment:
I might be missing something, but where is tree_level actually used here?

Reply:
Oh nvm, this is needed because tree_level is used by the helper.

Should we make it optional so other consumers of apply_to_modules don't have to add it to their API?

Reply:
it will be hard to make it optional based on current design, prefix is not used by consumers as well. But it is still a required signature, as it is used in the common traversal codes in "def f()"